public/docs/_examples/toh-5/dart/lib/dashboard_component.dart

```diff
@@ -0,0 +1,42 @@
+// #docplaster
+// #docregion
+import 'package:angular2/core.dart';
+// #docregion import-router
+import 'package:angular2/router.dart';
+// #enddocregion import-router
+
+import 'hero.dart';
+import 'hero_service.dart';
+
+@Component(
+  selector: 'my-dashboard',
+  // #docregion template-url
+  templateUrl: 'dashboard_component.html',
+  // #enddocregion template-url
+  // #docregion css
+  styleUrls: const ['dashboard_component.css']
+  // #enddocregion css
+)
+// #docregion component
+class DashboardComponent implements OnInit {
+// #docregion ctor
+  DashboardComponent(this._heroService, this._router);
+
+  Router _router;
+
+  HeroService _heroService;
+// #enddocregion ctor
+
+  List<Hero> heroes = [];
+
+  ngOnInit() async =>
+      heroes = (await _heroService.getHeroes()).getRange(1, 5);
+
+  // #docregion goto-detail
+  gotoDetail(Hero hero) {
```

Comment:
return type? Are there any guidelines about return types of lifecycle callbacks (`ngOnInit()`?